src/ble/BleLayer.cpp

```diff
@@ -203,9 +203,9 @@ CHIP_ERROR BleTransportCapabilitiesRequestMessage::Decode(const PacketBufferHand
     VerifyOrReturnError(CAPABILITIES_MSG_CHECK_BYTE_1 == chip::Encoding::Read8(p), BLE_ERROR_INVALID_MESSAGE);
     VerifyOrReturnError(CAPABILITIES_MSG_CHECK_BYTE_2 == chip::Encoding::Read8(p), BLE_ERROR_INVALID_MESSAGE);
 
-    for (size_t i = 0; i < kCapabilitiesRequestSupportedVersionsLength; i++)
+    for (unsigned char & mSupportedProtocolVersion : msg.mSupportedProtocolVersions)
```

Comment:
```suggestion
    for (unsigned char & supportedProtocolVersion : msg.mSupportedProtocolVersions)
```
But also, what guarantees that `sizeof(msg.mSupportedProtocolVersions) == kCapabilitiesRequestSupportedVersionsLength`?  They are not obviously the same size.  We need a static_assert at least.

Reply:
I think this for loop conversion was wrong. Will revert it back.

Reply:
Actually added the assert - I was looking at a different constant when comparing the logic. The automated conversion seems accurate.